atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/InstrumentedCassandraClient.java

```diff
@@ -63,14 +69,21 @@ public void batch_mutate(
             });
         });
 
-        tablesToCells.forEach(this::updateCellsWrittenForTable);
+        tablesToCells.forEach(this::maybeUpdateCellsWrittenForTable);
+    }
+
+    private void maybeUpdateCellsWrittenForTable(String table, Long numberOfCells) {
+        String tableTag = LoggingArgs.safeInternalTableNameOrPlaceholder(table);
+        if (log.isDebugEnabled() || cellsWrittenMetricTopListFilter.updateAndGetStatus(tableTag, numberOfCells)) {
+            updateCellsWrittenForTable(tableTag, numberOfCells);
```

Comment:
I think there are a couple of things to check with the semantics:
- calls to this method are made with the number of cells written to in a _single batch mutate operation_ and we use max as the merging function; this means that if we have a table P with 1000 batch mutates that write one cell and Q has one batch mutate that writes two cells, Q is prioritised above P, which I don't think is what we want? (This could be because P was written to across multiple transactions, or P is well distributed while Q hotspots.)
- if a table receives a large burst of writes every three minutes, it will flicker in and out of existence as far as the toplist is concerned (also, if the table also receives a steady stream of writes, writes during minutes where the table is not in the toplist won't count to its metric at all).